packages/flutter_tools/templates/app/windows.tmpl/runner/main.cpp.tmpl

```diff
@@ -21,6 +21,24 @@ int APIENTRY wWinMain(_In_ HINSTANCE instance, _In_opt_ HINSTANCE prev,
   RunLoop run_loop;
 
   flutter::DartProject project(L"data");
+
+  // Convert the UTF-16 command line arguments to UTF-8 for the Engine to use.
+  int argc, length;
+  wchar_t** argv = CommandLineToArgvW(command_line, &argc);
+  std::vector<std::string> command_line_arguments;
+
+  // Skip the first argument as it's the binary name.
+  for (int i = 1; i < argc; i++) {
+    length = WideCharToMultiByte(CP_UTF8, WC_NO_BEST_FIT_CHARS, argv[i], -1, nullptr, 0, nullptr, nullptr);
+    std::string arg(length, 0);
+    WideCharToMultiByte(CP_UTF8, WC_NO_BEST_FIT_CHARS, argv[i], -1, &arg[0], length, nullptr, nullptr);
+    command_line_arguments.push_back(std::move(arg));
+  }
+
+  LocalFree(argv);
```

Comment:
I think it's worth extracting all of this new code to a static helper function in the file that returns the vector; since we expect people to edit this file (especially main, as the template is currently structured), having things manual frees floating around in main itself seems like an unnecessary foot-gun.